config.yaml

```diff
@@ -77,3 +77,10 @@ options:
       A space-separated list of virtual IPs that the charm can utilize in conjunction with the
       HA Cluster subordinate charm. This option is mutually exclusive with the `ha-cluster-dns`
       option.
+  x-stream-protocol-version:
+    type: string
+    default: "v5.channel.k8s.io, v4.channel.k8s.io, v3.channel.k8s.io, v2.channel.k8s.io, channel.k8s.io"
+    destription: |
+       This option specifies the value of the X-Stream-Protocol-Version header.
+       Since nginx cannot merge multiple X-Stream-Protocol-Version headers,
+       this setting allows for manual specification to ensure proper transmission to the backend.
```

Comment:
```suggestion
       This option specifies the value of the X-Stream-Protocol-Version header.
       Since nginx cannot merge multiple X-Stream-Protocol-Version headers,
       this setting allows for manual specification to ensure proper transmission to the backend.
       The format should follow the RFC9110 specification, i.e., a comma separated list of the header
       values. https://datatracker.ietf.org/doc/html/rfc9110#name-field-lines-and-combined-fi
```

Reply:
Lovely.  Thought i'm not so super happy about a strict string that goes into the header either. 

I wouldn't mind this being a "space-separated" config, that the charm parses, confirms the values, and then applies to the header as the application requires today and tomorrow.

Reply:
@addyess I think I prefer your suggestion. It offloads the validation to the charm instead of directly setting the value in the config.

Reply:
done